src/v0/destinations/clevertap/transform.js

```diff
@@ -377,9 +383,48 @@ const processEvent = (message, destination) => {
 
 const process = (event) => processEvent(event.message, event.destination);
 
-const processRouterDest = async (inputs, reqMetadata) => {
-  const respList = await simpleProcessRouterDest(inputs, process, reqMetadata);
-  return respList;
+const processRouterDest = (inputs, reqMetadata) => {
+  // const respList = await simpleProcessRouterDest(inputs, process, reqMetadata);
+  // return respList;
+  const errorRespEvents = checkInvalidRtTfEvents(inputs);
+  if (errorRespEvents.length > 0) {
+    return errorRespEvents;
+  }
+
+  const eventsChunk = [];
+  const errorRespList = [];
+  // const { destination } = inputs[0];
+
+  inputs.forEach((event) => {
+    try {
+      let resp = event.message;
+      if (!event?.message?.statusCode) {
+        // already transformed event
+        resp = process(event);
+      }
+      eventsChunk.push({
+        message: Array.isArray(resp) ? [...resp] : resp,
+        metadata: event.metadata,
+        destination: event.destination,
+      });
+    } catch (error) {
```

Comment:
@yashasvibajpai take backlog item to cover failure scenario test cases too

Reply:
Added for myself